pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -180,6 +231,114 @@ func createRawWorkerDeployment(componentMeta metav1.ObjectMeta,
 	return deployment
 }
 
+func GetKServeContainerPort(podSpec *corev1.PodSpec) string {
+	for _, container := range podSpec.Containers {
+		if container.Name == "kserve-container" {
+			if len(container.Ports) > 0 {
+				return strconv.Itoa(int(container.Ports[0].ContainerPort))
+			}
+		}
+	}
+	return ""
+}
+func generateOauthProxyContainer(clientset kubernetes.Interface, isvc string, namespace string, upstreamPort string) (corev1.Container, error) {
+	configMap, err := clientset.CoreV1().ConfigMaps(constants.KServeNamespace).Get(context.TODO(), constants.InferenceServiceConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return corev1.Container{}, err
+	}
+	oauthProxyJSON := strings.TrimSpace(configMap.Data["oauthProxy"])
+	oauthProxyConfig := v1beta1.OauthConfig{}
+	if err := json.Unmarshal([]byte(oauthProxyJSON), &oauthProxyConfig); err != nil {
+		return corev1.Container{}, err
+	}
+	oauthImage := oauthProxyConfig.Image
+	oauthMemoryRequest := oauthProxyConfig.MemoryRequest
+	oauthMemoryLimit := oauthProxyConfig.MemoryLimit
+	oauthCpuRequest := oauthProxyConfig.CpuRequest
+	oauthCpuLimit := oauthProxyConfig.CpuLimit
```

Comment:
Looks like you've chosen to remove fallback values and the configurations in the ConfigMap are now mandatory. This is fine.

But, now, you need to validate that the configs are not empty. Please, do the validation and return errors appropriately.